src/main/java/com/facebook/openwifirrm/modules/Modeler.java

```diff
@@ -340,6 +343,57 @@ private void processData(InputData data) {
 		}
 	}
 
+	/**
+	 * NOTE: if a BSSID does not have a non-obsolete entry, it will be returned
+	 * (i.e., it will not be a key in the returned map).
+	 */
+	public Map<String, WifiScanEntry> getAggregatedWifiScans(long obsoletion_period) {
+		Map<String, WifiScanEntry> aggregatedWifiScans = new HashMap<>();
+		for (String bssid : dataModel.latestWifiScans.keySet()) {
+			// flatten the wifiscan entries and sort in reverse chronological order
+			List<List<WifiScanEntry>> scans = dataModel.latestWifiScans.get(bssid);
+			List<WifiScanEntry> mostRecentToOldest = scans.stream().flatMap(list -> list.stream())
+					.sorted((entry1, entry2) -> {
+						return -Long.signum(entry1.tsf - entry2.tsf);
+					}).collect(Collectors.toUnmodifiableList());
+
+			/*
+			 * For a given BSSID, discard the obsolete entries. Then,consider the most
+			 * recent entry. Take its ht_oper and vht_oper. Discard earlier entries with a
+			 * different ht_oper or a different vht_oper. For earlier entries with the same
+			 * ht_oper and vht_oper, take the average signal value.
+			 */
+			long now = Instant.now().getEpochSecond();
```

Comment:
I think you want higher resolution here, e.g. for milliseconds use `System.currentTimeMillis()`. Also, typically `now` should be defined outside the loop unless there is a good reason not to.

Reply:
Now should be defined outside the loop, yes. I'm still looking into confirming whether there is some reference/start for tsf. I'm almost certain the units are microseconds, but I don't know when "time = 0" is. As Jeffrey says in a different comment, we may have to store the time ourselves.

Reply:
We don't need to know time = 0 actually. We just need to compare the latest tsf and the tsf of the other entries. Would you mind pointing me to where @elludraon said to store the time ourselves? Thanks.